es6/docxtemplater.js

```diff
@@ -37,16 +37,40 @@ const Docxtemplater = class Docxtemplater {
 		this.compiled = {};
 		this.modules = [commonModule()];
 		this.setOptions(options);
-		modules.forEach(module => {
-			this.attachModule(module);
+		modules.forEach(mod => {
+			this.attachModule(mod);
 		});
 		if (zip) {
 			if (!zip.files || typeof zip.file !== "function") {
 				throw new Error(
 					"The first argument of docxtemplater's constructor must be a valid zip file (jszip v2 or pizzip v3)"
 				);
 			}
-			this.loadZip(zip).compile();
+			this.loadZip(zip);
+			const detach = (mod) => {
+				const indexOfModule = this.modules.indexOf(mod);
+				// module will always be present in the this.modules, hence we are not validating index.
+				const removedModule = this.modules.splice(indexOfModule, 1)[0];
+				removedModule.attached = false;
```

Comment:
Please do not set attached to false in this case, as I want to discourage users from reusing multiple times the same module.

Reply:
👍

Reply:
Done. All detach is doing now is sending the event.